File: cpp_src/spawn_turtle/src/spawn_turtle.cpp
#include <ros/ros.h>
#include <turtlesim/Spawn.h>

int main(int argc, char ** argv) {
    ros::init(argc, argv, "spawn_turtle");
    ros::NodeHandle nh;

    ros::ServiceClient spawnClient = nh.serviceClient<turtlesim::Spawn>("spawn");

    ros::service::waitForService("spawn");

    for (int i = 0; i < 10; i ++) {
        turtlesim::Spawn::Request req;
        turtlesim::Spawn::Response resp;

        req.x = 1 + i;
        req.y = 1 + i;
        req.theta = M_PI/2;
        req.name = "Leo" + std::to_string(i);

        bool ok = spawnClient.call(req, resp);
        if (ok) {
            ROS_INFO("spawned a turtle named %s", resp.name.c_str());
        } else {
            ROS_ERROR("Failed to spawn");
        }

    }


    return 0;
}

File: pyhon_src/spawn_turtle/scripts/spawn_turtle.py
#! /usr/bin/env python3

import rospy
import math
from turtlesim.srv import Spawn, SpawnRequest


def main():
    rospy.init_node("spawn_turtle")
    client = rospy.ServiceProxy("spawn", Spawn)
    rospy.wait_for_service("spawn")

    for i in range(10):
        try:

            req = SpawnRequest()
            req.x = 1 + i
            req.y = 1 + i
            req.theta = math.pi/2
            req.name = "Leo%s" %i

            resp = client(req)
            rospy.loginfo("spawned a turtle name %s" %resp.name)
        except rospy.ServiceException as e:
            rospy.loginfo("service call fail: %s" %e)

if __name__ == "__main__":
    main()